ext/sqlite3/database.c

```diff
@@ -17,6 +17,8 @@ database_mark(void *ctx)
 {
     sqlite3RubyPtr c = (sqlite3RubyPtr)ctx;
     rb_gc_mark(c->busy_handler);
+    rb_gc_mark(c->stmt_timeout);
+    rb_gc_mark(c->stmt_deadline);
```

Comment:
You only need to mark Ruby object references (`VALUE`).

```suggestion
```

Reply:
Ah, ok, improving my mental model here.